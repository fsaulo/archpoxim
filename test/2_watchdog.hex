0xDC00000E
0xFC000000
0xFC000000
0xFC000000
0xDC000003
0x00000000
0x00000000
0x00000000
0x083A0000
0x085B0000
0x68600017
0x14011800
0xD0000001
0xFC000064
0x80000000
0x03C07FFC
0x87E10001
0x68200018
0x68400019
0x10021701
0x74220000
0xDFFFFFFF
0xFC000000
0xE1AC04DA
0x80000064
0x80808080
