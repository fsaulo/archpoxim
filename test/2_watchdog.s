// Segmento de codigo
.text
    // Tabela de vetor de interrupcao
    init:
        bun main
        int 0
        int 0
        int 0
        bun isr_hw1
        .align 5
    // Rotina de tratamento de interrupcao
    isr_hw1:
        // Checagem de codigo de interrupcao
        mov r1, cr
        mov r2, ipc
        l32 r3, [codigo]
        cmp r1, r3
        bne 1
        // Interrupcao de software 100
        int 100
        // Retorno de interrupcao
        reti
    // Funcao principal
    main:
        // SP = 32 KiB
        mov sp, 0x7FFC
        // Habilitando interrupcao (IE = 1)
        sbr sr[1]
        // R1 = valor do contador
        l32 r1, [valor]
        // R2 = endereco do contador
        l32 r2, [watchdog]
        sra r2, r2, 2
        // Watchdog = R1
        s32 [r2], r1
        // Laco infinito
        bun -1
        // Finalizacao de execucao
        int 0
// Segmento de dados
.data
    // Codigo de interrupcao
    codigo:
        .4byte 0xE1AC04DA
    // Valor do contador
    valor:
        .4byte 0x80000064
    // Endereco do dispositivo
    watchdog:
        .4byte 0x80808080
