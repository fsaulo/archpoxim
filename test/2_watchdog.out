[START OF SIMULATION]
0x00000000:	bun 14                   	PC=0x0000003C
0x0000003C:	mov sp,32764             	SP=0x00007FFC
0x00000040:	sbr sr[1]                	SR=0x00000002
0x00000044:	l32 r1,[r0+24]           	R1=MEM[0x00000060]=0x80000064
0x00000048:	l32 r2,[r0+25]           	R2=MEM[0x00000064]=0x80808080
0x0000004C:	sra r0,r2,r2,1           	R0:R2=R0:R2>>2=0x0000000020202020,SR=0x00000002
0x00000050:	s32 [r2+0],r1            	MEM[0x80808080]=R1=0x80000064
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
0x00000054:	bun -1                   	PC=0x00000054
[HARDWARE INTERRUPTION 1]
0x00000010:	bun 3                    	PC=0x00000020
0x00000020:	add r1,cr,r0             	R1=CR+R0=0xE1AC04DA,SR=0x00000012
0x00000024:	add r2,ipc,r0            	R2=IPC+R0=0x00000054,SR=0x00000002
0x00000028:	l32 r3,[r0+23]           	R3=MEM[0x0000005C]=0xE1AC04DA
0x0000002C:	cmp r1,r3                	SR=0x00000042
0x00000030:	bne 1                    	PC=0x00000034
0x00000034:	int 100                  	CR=0x00000064,PC=0x0000000C
[SOFTWARE INTERRUPTION]
0x0000000C:	int 0                    	CR=0x00000000,PC=0x00000000
[END OF SIMULATION]
